File: dandanplay_native/macos/ios_mac_share/Model/Message/Receive/HUDMessage.swift
//
//  HUDMessage.swift
//  DanDanPlayExperience
//
//  Created by JimHuang on 2020/2/3.
//  Copyright © 2020 JimHuang. All rights reserved.
//

import HandyJSON

class HUDMessage: BaseModel {
    enum Style: String, HandyJSONEnum {
        case tips
        case progress
    }
    
    var style = Style.tips
    var text = ""
    var progress: Float = 0.0
    var isDismiss = false
    var key = ""
}

File: dandanplay/lib/Model/Message/Send/HUDMessage.dart

import 'package:dandanplay/Model/Message/Send/BaseMessage.dart';
import 'package:flutter/cupertino.dart';
import 'package:json_annotation/json_annotation.dart';

part 'HUDMessage.g.dart';

enum HUDMessageStyle {
  tips,
  progress
}

@JsonSerializable()
class HUDMessage extends BaseMessage {

  @JsonKey(ignore: true)
  @override
  String get name => "HUDMessage";

  @override
  Map<String, dynamic> get data {
    return _$HUDMessageToJson(this);
  }

  HUDMessageStyle style;
  String text;
  double progress = 0;
  bool isDismiss = false;
  String key;

  HUDMessage({@required this.style, this.text, this.progress, this.isDismiss, this.key});

}